executor/write.go

```diff
@@ -158,8 +158,13 @@ func updateRecord(ctx sessionctx.Context, h int64, oldData, newData []types.Datu
 	} else {
 		sc.AddAffectedRows(1)
 	}
-
-	ctx.GetSessionVars().TxnCtx.UpdateDeltaForTable(t.Meta().ID, 0, 1)
+	colSize := make(map[int64]int64)
+	for id, col := range t.WritableCols() {
+		if col.State == model.StatePublic && newData[id].Size() != 0 {
+			colSize[col.ID] = int64(newData[id].Size() - oldData[id].Size())
```

Comment:
This is not right? If the new data happens to be null, but the old data is not null.